### DISCUSSION TITLE
Issues with plotting

### DISCUSSION THREAD

**Discussion by waffvil** [observablehq/plot#2226] [Q&A] [answered]
I have been trying to plot some data. this is my code. I am using the observable framework so I can save it on vscode and it will update my website. When i update my code and save it it turns out fine it looks good. But when I refresh the page or go back to it. The graph just changes. Shown in the images

```js
Plot.dot(df, {
    width: 1000,
    height: 500,
    x: "Transaction.Date",
    y: "Debit.Amount",
    //fill: "Category",
    //stroke: "Category",
    channels: {x: "Transaction.Date", y: "Debit.Amount"},
    tip: true,
}).plot()
```

<img width="341" alt="Screenshot 2024-11-12 at 07 38 55" src="https://github.com/user-attachments/assets/200d2e37-98b8-4f6a-8f09-7d5b8a695ab7">
<img width="345" alt="Screenshot 2024-11-12 at 07 40 12" src="https://github.com/user-attachments/assets/4d92e33a-61ec-4d7c-8dc8-3c6bd29e6fae">

**Comment by Fil** [comment, marked as answer]
Do you by any chance have some code that mutates the `df` variable as a side effect? This might explain the discrepancy.

**Reply by waffvil** [reply]
omg, thank you so much, yeah I was directly changing my main df, ive assigned it to another df and it works perfectly fine now. Massive thanks!!
